Tasks/Common/webdeployment-common-v2/ziputility.ts

```diff
@@ -2,13 +2,36 @@ import tl = require('azure-pipelines-task-lib/task');
 import path = require('path');
 import Q = require('q');
 import fs = require('fs');
-import { ToolRunner } from 'azure-pipelines-task-lib/toolrunner';
 
 var DecompressZip = require('decompress-zip');
 var archiver = require('archiver');
 
 const deleteDir = (path: string) => tl.exist(path) && tl.rmRF(path);
 
+const extractWindowsZip = async (fromFile: string, toDir: string, usePowerShell?: boolean) => {
+    if (usePowerShell) {
+        await extractUsingPowerShell(fromFile, toDir);
+    }
+    else {
+        await extractUsing7zip(fromFile, toDir);
+    }
+}
+
+const extractUsingPowerShell = async (fromFile: string, toDir: string) => {
+    tl.debug(`Using PowerShell for extracting zip ${fromFile}`);
+    let command = `Expand-Archive -Path "${fromFile}" -DestinationPath "${toDir}"`;
+    await tl.exec(`powershell.exe`, [
```

Comment:
We shouldn't be using `powershell.exe` directly but as a fallback like below:
```powershell
let powershell = tl.tool(tl.which('pwsh') || tl.which('powershell') || tl.which('pwsh', true))
```
Have a look [here](https://github.com/microsoft/azure-pipelines-tasks/blob/8f6b03510319518c15674da50d8f91561d68fb6a/Tasks/PowerShellV2/powershell.ts#L94) for reference.

Reply:
because we are doing this only when the platform on which the task is running is 'Windows', so we are sure to find 'powershell.exe' there and so don't have to look for pwsh.

Reply:
@arjgupta have a look at this merged commit, the comment in code says so. You might want to check with the commit owner for reasoning if needed. Most likely there might be plans to remove it from windows as well and make use of pwsh across platforms.